490.py

```diff
@@ -0,0 +1,73 @@
+# class Solution:
+#     def hasPath(self, maze: List[List[int]], start: List[int], destination: List[int]) -> bool:
+#         self.R, self.C = len(maze), len(maze[0])
+#         visited = [[False] * self.C for _ in range(self.R)]
+
+#         return self.dfs(maze, start, destination, visited)
+
+#     def dfs(self, maze: List[List[int]], start: List[int], destination: List[int], visited: List[List[int]]) -> bool:
+#         sr, sc = start[0], start[1]
+
+#         if visited[sr][sc]:
+#             return False
+
+#         if sr == destination[0] and sc == destination[1]:
+#             return True
+
+#         visited[sr][sc] = True
+#         up, down, left, right = sr - 1, sr + 1, sc - 1, sc + 1
+
+#         while up >= 0 and maze[up][sc] == 0:
+#             up -= 1
+
+#         if (self.dfs(maze, [up + 1, sc], destination, visited)):
+#             return True
+
+#         while down < self.R and maze[down][sc] == 0:
+#             down += 1
+
+#         if (self.dfs(maze, [down - 1, sc], destination, visited)):
+#             return True
+
+#         while left >= 0 and maze[sr][left] == 0:
+#             left -= 1
+
+#         if (self.dfs(maze, [sr, left + 1], destination, visited)):
+#             return True
+
+#         while right < self.C and maze[sr][right] == 0:
+#             right += 1
+
+#         if (self.dfs(maze, [sr, right - 1], destination, visited)):
+#             return True
+
+#         return False
+# DFS solution.
+
+
+class Solution:
+    def hasPath(self, maze: List[List[int]], start: List[int], destination: List[int]) -> bool:
+        R, C = len(maze), len(maze[0])
+        visited = [[False] * C for _ in range(R)]
+        DIRS = ((1, 0), (-1, 0), (0, 1), (0, -1))
+        queue = deque([start])
+        visited[start[0]][start[1]] = True
```

Comment:
I strongly object to this `visited` name here. Technically, when you mark a cell to `True`, you have **_not_** "visited" this cell yet; you just discovered this cell and put it into a queue. On Wikipedia, the BFS pseudo code uses `discovered` to name this variable and the LeetCode community tend to use `seen`, both of which are okay with me.